### ISSUE TITLE
Unable to start Vagrant Developer VM in Kubernetes mode

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#5082]
I have a branch checked out with minimal changes from master (I can try reproducing this on master if needed)

At the root of the Cilium repository, run the following:
```
NFS=1 K8S=1 ./contrib/vagrant/start.sh
```

The provisioning step fails:
```
    k8s1: [Service]
    k8s1: ExecStart=/usr/bin/kube-proxy \
    k8s1:   --cluster-cidr=10.0.0.0/10 \
    k8s1:   --kubeconfig=/var/lib/kube-proxy/kube-proxy.kubeconfig \
    k8s1:   --proxy-mode=iptables \
    k8s1:   --v=2
    k8s1: 
    k8s1: Restart=on-failure
    k8s1: RestartSec=5
    k8s1: 
    k8s1: [Install]
    k8s1: WantedBy=multi-user.target
    k8s1: Created symlink /etc/systemd/system/multi-user.target.wants/kube-proxy.service -> /etc/systemd/system/kube-proxy.service.
    k8s1: Failed to restart kube-proxy.service: Unit containerd.service not found.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by ianvernon** [comment]
I am going to see if this was affected by the recent update of the Vagrant box.
I have reverted the following commits in a private branch:

* 31d053c64abc581bb43c4c744ce51861afefbaea
* f39f475d2ed5eb5d0ac03f0f38db2616b6bcb072
* 7c5333b925a64feb93b394d45e44eab2d1d248a9
* bbfcb3e97372e5ab1946a88d257fdd8aa938664a

**Comment by ianvernon** [comment]
Reverting the above commits remedied the issue. It appears that the Vagrant box version update removed containerd, which is a prerequisite for kube-proxy?

**Comment by joestringer** [comment]
@ianvernon It's expected that docker-ce now installs this, it was causing trouble due to conflicts:

https://github.com/cilium/packer-ci-build/pull/77

See also this Slack thread:

https://cilium.slack.com/archives/C2B917YHE/p1532986793000342

(I don't know the details, you'd have to sync with @rlenglet  / @eloycoto ).

**Comment by ianvernon** [comment]
Marking this as high priority because this significantly impacts my development workflow.

**Comment by rlenglet** [comment]
I'd suggest defining a dummy `containerd.service` that does nothing.

**Comment by rlenglet** [comment]
Or we could override the `kube-proxy` unit to override the dependency to remove the dependency on `containerd.service`. Cf. https://coreos.com/os/docs/latest/using-systemd-drop-in-units.html
But that might be tricky:
https://blog.inf.ed.ac.uk/sl7project/2014/12/05/systemd-issues-with-overriding-dependencies/

**Comment by ianvernon** [comment]
Fixed-by #5149; closing.
